Content/Passives/Ranged/Masteries/ContagionBurstMastery.cs

```diff
@@ -0,0 +1,55 @@
+﻿using PathOfTerraria.Common.Systems.PassiveTreeSystem;
+using PathOfTerraria.Content.Buffs.ElementalBuffs;
+using System.Collections.Generic;
+using System.Reflection;
+
+namespace PathOfTerraria.Content.Passives;
+
+internal class ContagionBurstMastery : Passive
+{
+	internal class ContagionBurstPlayer : ModPlayer
+	{
+		private delegate void hook_OnHitNPC(Player player, NPC npc, in NPC.HitInfo hit, int damageDone);
+
+		public override void Load()
+		{
+			// We need to apply this after the current NPC takes debuffs on OnHitNPC.
+			// This makes it feel more consistent for the player, as an NPC would have the 'visible' amount of stacks applied.
+			MonoModHooks.Add(typeof(PlayerLoader).GetMethod(nameof(PlayerLoader.OnHitNPC), BindingFlags.Static | BindingFlags.Public), LateOnHitNPC);
+		}
+
+		private static void LateOnHitNPC(hook_OnHitNPC orig, Player player, NPC npc, in NPC.HitInfo hit, int damageDone)
+		{
+			orig(player, npc, in hit, damageDone);
+
+			if (player.GetModPlayer<PassiveTreePlayer>().TryGetCumulativeValue<ContagionBurstMastery>(out float value) 
+				&& npc.GetGlobalNPC<PoisonNPC>().Stacks is { Count: >0 } stacks && npc.life <= 0)
+			{
+				int stacksToAdd = (int)MathF.Ceiling(stacks.Count * value / 100f);
+
+				foreach (NPC other in Main.ActiveNPCs)
+				{
+					if (other.CanBeChasedBy() && other.whoAmI != npc.whoAmI && other.DistanceSQ(npc.Center) < PoTMod.NearbyDistanceSq)
+					{
+						List<PoisonNPC.PoisonStack> list = other.GetGlobalNPC<PoisonNPC>().Stacks;
+						int time = 0;
+
+						for (int i = 0; i < stacksToAdd; ++i)
+						{
+							PoisonNPC.PoisonStack stack = stacks[i];
+							list.Add(stack);
+							time = Math.Max(time, stack.MaxTime);
+						}
+
+						other.AddBuff(ModContent.BuffType<PoisonedDebuff>(), time);
```

Comment:
Stacks are a by-ref class, yet you are distributing the same instance across multiple enemies, causing these enemies to decay that instance multiple times per tick, no? If there is no code that keeps and checks on a stack's handle, then you should make these stacks into record structs.

Additionally I think you should make the `Stacks` list field private, instead providing a `ReadOnlySpan<...> Stacks => CollectionsMarshal.AsSpan(stacks)` property and public methods for adding and copying stacks, just to centralize mutation and reduce coupling. But that's nitpicking.